OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -88,9 +88,10 @@ void PhysicalFrame::extendConnectToModel(Model& aModel)
 {
     Super::extendConnectToModel(aModel);
 
-    // TODO: Better use name search or more robust method
-    if (upd_geometry(0).getFrameName() == "")
-        upd_geometry(0).setFrameName(getName());
+    // 
+    int idx = getProperty_geometry().findIndexForName("frame_geometry");
+    if (idx != -1 && upd_geometry(idx).getFrameName() == "")
+        upd_geometry(idx).setFrameName(getName());
```

Comment:
@sherm1 I wasn't so concerned because I think @aseth1 is interested in having frames visualize themselves via `generateDecorations()`, and that `frame_geometry` and related code is temporary until that happens.